corehq/apps/app_manager/xform.py

```diff
@@ -1276,6 +1276,15 @@ def make_delegation_stub_case_block():
                 )
                 if 'external_id' in actions['open_case'] and actions['open_case'].external_id:
                     extra_updates['external_id'] = actions['open_case'].external_id
+            elif module.parent_select.active:
+                # This is a submodule. case_id will have changed to avoid a clash with the parent case.
+                # Case type is enough to ensure uniqueness for normal forms. No need to worry about a suffix.
+                case_id = '_'.join((CASE_ID, form.get_case_type()))
```

Comment:
After all the work done in suite_xml.py this feels like cheating. OR is there a way we can shortcut the algorithm in suite_xml.py?